api/convert/teleport/accesslist/v1/accesslist.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accesslistv1
```

Comment:
Do we really want to have a separate package for convert functions ? 

Right now most of conv function are located in api/type like in case of [DeviceFromResource](https://github.com/gravitational/teleport/blob/52e357f1cccaee3320fc35bcedd64bf269836546/api/types/device.go#L89)

I think that we should keep it consistence or refactor/move existing convert methods.  By introduction of in `convert/teleport` in current form will make things more complicated and hard to find.

Reply:
I wasn't 100% sure on this. However, there's an advantage to keeping them separate in that we don't have to import all of the proto messages when importing api/types -- what do you think about adding a package in there?

Reply:
I'd like to note here that I was thinking specifically of splitting out servicecfg and the impact it had on the resulting Teleport binary sizes. I think we have the opportunity to make a similar change here by having a separate conversion package be the only window into the protobuf messages where necessary.

Reply:
Sorry, I just re-read your blurb -- one thing here is that this is attempting to introduce a new way of handling proto objects. The eventual approach will be that everything is one way or the other, but I don't want to balloon this feature with a major refactor as I'm kind of using it to help define what the new way "should" be.

Reply:
Okay, I've done a fair bit of refactoring here. I've moved the internal representations to `lib/types` and the conversion functions live close by. I've also renamed the conversions from `From/ToV1` to `From/ToProto` sitting in relevant packages.